src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const splittedDate = date.split(separator);
+  const lengthSplitedDate = splittedDate.length;
+  const storeDateObj = {};
+
+  for (let i = 0; i < lengthSplitedDate; i++) {
+    const keyFromFormat = fromFormat[i];
+    const valueFromSplitteDate = splittedDate[i];
+
+    if (keyFromFormat === 'YYYY') {
+      storeDateObj['YY'] = valueFromSplitteDate.slice(2);
+    }
+
+    if (keyFromFormat === 'YY') {
+      if (+valueFromSplitteDate < 30) {
+        storeDateObj['YYYY'] = '20' + valueFromSplitteDate;
+      } else {
+        storeDateObj['YYYY'] = '19' + valueFromSplitteDate;
+      }
+    }
+
+    storeDateObj[keyFromFormat] = valueFromSplitteDate;
```

Comment:
Maybe you should check if this is a month or a day, but not a year, because you already assigned values for year (your code line 62-72).

Reply:
no need i think,  because we have sure keys (formats) in input datas (they are in the conditions of task. in case Datas will bad we will never understand what is year and what is month and what is day).